pkg/controller/apimanagerbackup/apimanagerbackup_logic_reconciler.go

```diff
@@ -0,0 +1,398 @@
+	if res.Requeue || err != nil {
+		return res, err
+	}
+
+	res, err = r.reconcileBackupSystemFileStoragePVCToPVCJob()
+	if res.Requeue || err != nil {
+		return res, err
+	}
+
+	return res, err
+}
+
+func (r *APIManagerBackupLogicReconciler) reconcileBackupDestinationPVC() error {
+	desired := r.apiManagerBackup.BackupDestinationPVC()
+	if desired == nil {
+		return nil
+	}
+
+	// TODO create mutator function for PVC ?
+	err := r.ReconcileResource(&v1.PersistentVolumeClaim{}, desired, reconcilers.CreateOnlyMutator)
+
+	return err
+}
+
+func (r *APIManagerBackupLogicReconciler) reconcileBackupSecretsAndConfigMapsToPVCJob() (reconcile.Result, error) {
+	desired := r.apiManagerBackup.BackupSecretsAndConfigMapsToPVCJob()
+	if desired == nil {
+		return reconcile.Result{}, nil
+	}
+
+	// TODO create Mutator function for Jobs??
+	existing := &batchv1.Job{}
+	// Check if this step has finished
+	if r.cr.SecretsAndConfigMapsBackupStepFinished() {
+		return reconcile.Result{}, nil
+	}
+
+	// Check if backup substep has completed
+	if !r.cr.SecretsAndConfigMapsBackupSubStepFinished() {
+		err := r.ReconcileResource(existing, desired, reconcilers.CreateOnlyMutator)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		substepFinished := true
+		if existing.Status.Succeeded != *desired.Spec.Completions {
+			r.Logger().Info("Job has still not finished", "Job Name", desired.Name, "Actively running Pods", existing.Status.Active, "Failed pods", existing.Status.Failed)
+			return reconcile.Result{Requeue: true}, nil
+		}
+
+		r.cr.Status.SecretsAndConfigMapsBackupSubStepFinished = &substepFinished
+		err = r.UpdateResourceStatus(r.cr)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		r.Logger().Info("Job finished successfully. Requeing", "Job Name", desired.Name)
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	// Check if cleanup substep has completed
+	if !r.cr.SecretsAndConfigMapsCleanupSubStepFinished() {
+		common.TagToObjectDeleteWithPropagationPolicy(desired, metav1.DeletePropagationForeground)
+		err := r.ReconcileResource(existing, desired, reconcilers.CreateOnlyMutator)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		// TODO is this needed or it's somehow handled correctly in ReconcileResource?
+		// The behavior we want is that we only update the state of the CR when we are
+		// sure the object does not exist anymore. The reason of that is that
+		// if we do not do it this way it could happen that we mark the state of this
+		// substep as completed but the Job is not completely deleted and we move to the
+		// next step, which would fail because Job is not deleted nor the PVC associated
+		// to it, which would provoke an error when trying to reuse it for the next job PVC
+		err = r.GetResource(common.ObjectKey(desired), existing)
+		if err != nil && !errors.IsNotFound(err) {
+			return reconcile.Result{}, err
+		}
+
+		substepFinished := true
```

Comment:
Here the job might still be alive, right? You do not check it does not exists.

Reply:
You are right. I've added check here and on other places where the deletion is performed.